### PR TITLE
add option to create new list from "add to list" dialog

### LINKED ISSUE #2357: Implement "Add New List" button on "Add To List" Dialog
Triggering the 'Create New List' dialog from the 'Add to List' dialog did not work seamlessly as expected, only one dialog can be open at a time. Here is the approach suggested by @aliceriot:

- refactor the create user list form so that it's a self-contained component and not tied to the dialog
- clicking a 'Create List' button  will change what form is displayed in the add to list dialog to 'create new list' instead
- have a 'back' button up on the corner to back out of creating a new list(that would go back to the list of already-created lists)
- then if the user wanted to create a list they could create it there and then we could back them out to the initial screen in the dialog

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
  - [ ] Tag @ferdi or @pdpinch for review
- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?

closes #2357

#### What's this PR do?

this adds a button in the "add to list" dialog which lets you create a new list.

basically, it just opens the existing form we have for creating userlists.

#### How should this be manually tested?

Try it out and make sure there are no weird edges. you should be able to:

- add and remove a list from an existing list (i.e. no regression)
- clicking the 'Create New List' button should open the dialog
- canceling that dialog should just show you the 'add to list' dialog again
- you should be able to successfully create a new list
- that list should show up in the 'add to list' dialog immediately

I think that's about it. Just make sure it all makes sense and works properly.

#### Screenshots (if appropriate)

flooofdfasdfasdf
flooofdf
sdfasjdfansdfn

### CODE REVIEW

**Comment by aliceriot** [comment]
ahhh I forgot that was why we had to use the redux store for the object, haha

**Comment by aliceriot** [comment]
@mbertrand ok that issue should be fixed now
